metaworld_algorithms/nn/moore.py

```diff
@@ -3,15 +3,138 @@
 import jax
 import jax.numpy as jnp
 from jaxtyping import Array, Float
+from typing import Tuple
 
 from metaworld_algorithms.config.nn import MOOREConfig
 
 from .base import MLP
 
+# --- OPTIMIZED FUNCTION USING JAX.LAX.SCAN ---
 
 def orthogonal_1d(
     x: Float[Array, "batch_size num_experts dim"], num_experts: int
 ) -> Float[Array, "batch_size num_experts dim"]:
+    
```

Comment:
Why did this function need to be rewritten? Can you provide justification in the form of a script comparing the performance (and speed) between this and my original implementation?